### PR TITLE
Define the uniform neighbor sampling C API

### PR DESCRIPTION
**Pull request by ChuckHastings**
Define the uniform neighborhood sampling C API.  The implementation will be done in a separate PR.  An outline of what the implementation will resemble is provided (but ifdef'ed out).

### CODE REVIEW

File: cpp/src/c_api/uniform_neighbor_sampling.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  cugraph_type_erased_host_array_view_t const* fan_out_{nullptr};
+  bool with_replacement_{false};
+  cugraph_sample_result_t* result_{nullptr};
+
+  uniform_neighbor_sampling_functor(::cugraph_resource_handle_t const* handle,
+                                    ::cugraph_graph_t* graph,
+                                    ::cugraph_type_erased_device_array_view_t const* start,
+                                    ::cugraph_type_erased_device_array_view_t const* start_label,
+                                    ::cugraph_type_erased_host_array_view_t const* fan_out,
+                                    bool without_replacement)
```
Comment (aschaffer) [inline cpp/src/c_api/uniform_neighbor_sampling.cpp:63]: The C++ API defines the semantic of this flag as `with_replacement` (which defaults to `true`). I know this code is `ifdef`-out for now, but it's meant to be brought back, as far as I understand.

**Review by aschaffer — CHANGES_REQUESTED** [review]

File: cpp/include/cugraph_c/algorithms.h [hunk 2]
```diff
@@ -392,6 +392,93 @@ cugraph_type_erased_device_array_view_t* cugraph_random_walk_result_get_path_siz
...
+ * @param [out] error        Pointer to an error object storing details of any error.  Will
+ *                           be populated if error code is not CUGRAPH_SUCCESS
+ * @return error code
+ */
+cugraph_error_code_t uniform_nbr_sample(const cugraph_resource_handle_t* handle,
+                                        cugraph_graph_t* graph,
+                                        const cugraph_type_erased_device_array_view_t* start,
+                                        const cugraph_type_erased_device_array_view_t* start_label,
+                                        const cugraph_type_erased_host_array_view_t* fan_out,
+                                        bool_t without_replacement,
```
Comment (seunghwak) [inline cpp/include/cugraph_c/algorithms.h:426]: No `do_expensive_check` here (at least as a placeholder even if we're not checking anything at this moment)?

**Review by seunghwak — APPROVED** [review]

File: cpp/src/c_api/uniform_neighbor_sampling.cpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/src/c_api/uniform_neighbor_sampling.cpp:63]: Yes.  I defined this based on the old definition in the C++ API.  Fixed in latest push.

**Review by aschaffer — APPROVED** [review]

File: cpp/include/cugraph_c/algorithms.h (hunk 2 above)
Comment (ChuckHastings) [reply cpp/include/cugraph_c/algorithms.h:426]: Added in latest push.

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**